system/testing/BaseTestCase.cfc

```diff
@@ -485,4 +516,36 @@ component extends="testbox.system.compat.framework.TestCase" accessors="true"{
 		return new coldbox.system.core.util.Util();
 	}
 
-}
\ No newline at end of file
+    /**
+    * Separate a uri into two parts: the base uri, and a query string collection
+    *
+    * @return a struct containing the base uri and a struct of query string parameters
+    */
+    private struct function explodeURI( required string uri ){
+        var uriParts = ListToArray( arguments.uri, '?' );
+
+        var queryParams = {};
+        if ( ArrayLen( uriParts ) > 1 ){
+            var queryParamsArray = ListToArray( uriParts[2], '&' );
+
+            for ( var queryStringPair in queryParamsArray ){
+                var pairParts = ListToArray( queryStringPair, '=' );
+                if ( ! StructKeyExists( queryParams, pairParts[1] ) ){
+                    queryParams[ pairParts[ 1 ] ] = pairParts[ 2 ];
```

Comment:
I think this will error if you have empty value such as `?foo=` since `pairParts[ 2 ]` won't exist.